Weather.py

```diff
@@ -0,0 +1,281 @@
+''' Function Based Weather Man App '''
+
+import os
+import sys
+from collections import namedtuple
+
+# defined a named tuple for the columns in the data
+weather_data = namedtuple(
+        "weather_data",
+        [
+            "PKT",
+            "MaxTemperatureC",
+            "MeanTemperatureC",
+            "MinTemperatureC",
+            "DewPointC",
+            "MeanDewPointC",
+            "MinDewpointC",
+            "MaxHumidity",
+            "MeanHumidity",
+            "MinHumidity",
+            "MaxSeaLevelPressurehPa",
+            "MeanSeaLevelPressurehPa",
+            "MinSeaLevelPressurehPa",
+            "MaxVisibilityKm",
+            "MeanVisibilityKm",
+            "MinVisibilitykM",
+            "MaxWindSpeedKmh",
+            "MeanWindSpeedKmh",
+            "MaxGustSpeedKmh",
+            "Precipitationmm",
+            "CloudCover",
+            "Events",
+            "WindDirDegrees"
+        ]
+    )
+
+
+''' function that takes command line arguments and processes the files and
+folders for extracting data '''
+
+
+def process_folder(arguments):
+
+    data = []
+    files = os.listdir(arguments[2])
+
+    for file in files:
+        if arguments[1] in file:
+            with open(
+                    arguments[2]
+                    + "/"
+                    + file
+                    ) as file_object:
+                for line in file_object:
+                    temp_tuple = line.strip().split(",")
+                    if not temp_tuple or "<" in temp_tuple[0]:
+                        break
+                    if temp_tuple[0].isalpha() or temp_tuple == [""]:
+                        continue
+                    data.append(weather_data._make(temp_tuple))
+
+    return data
+
+
+''' task 01 to extract values of highest temperature and day, lowest
+temperature and day and highest humidity and day in an year '''
+
+
+def yearly_value_temperatures_and_humidity(data):
+
+    max_temp = float("-inf")
+    min_temp = float("inf")
+    max_humid = float("-inf")
+
+    def resolve_date(date):
+        date = date.split("-")
+        date[1] = mapping_months(int(date[1]))
+        date.pop(0)
+        s = " "
+        return s.join(date)
+
+    for extracted_tuple in data:
+
+        # finding maximum temperature
+        if extracted_tuple.MaxTemperatureC != "" and max_temp < \
+                int(extracted_tuple.MaxTemperatureC):
+            max_temp = int(extracted_tuple.MaxTemperatureC)
+            max_temp_day = resolve_date(extracted_tuple.PKT)
```

Comment:
better use float for type casting instead of integer for scalability purpose. This comment apply to whole application.